static/js/types/common.d.ts

```diff
@@ -0,0 +1,31 @@
+export interface LxdConfigPair {
+  [key: string]: string;
+}
+
+export interface LxdConfigOption {
+  key: string;
+  type: "string" | "integer" | "bool";
+  scope: "global" | "local";
+  default: string | boolean;
+  description: string;
+}
+
+export interface LxdDevice {
+  type: string;
+}
+
+export interface LxdDiskDevice extends LxdDevice {
+  path: string;
+  pool: string;
+  type: "disk";
+}
+
+export interface LxdNicDevice extends LxdDevice {
+  name: string;
+  network: string;
+  type: "nic";
+}
+
+export interface LxdDevices {
+  [key: string]: LxdDevice;
```

Comment:
Considering the use that I had in mind for the `LxdDevice` type, accessing only the `type` property was fine, but I agree that it's easier to think about this kind of generalisation later because it complicates things now. It is addressd in #102